client.go

```diff
@@ -0,0 +1,74 @@
+package filetypes
+
+import (
+	"github.com/cloudquery/filetypes/csv"
+	"github.com/cloudquery/filetypes/json"
+)
+
+type FormatType string
+
+const (
+	FormatTypeCSV  = "csv"
+	FormatTypeJSON = "json"
+)
+
+type FileSpec struct {
+	Format         FormatType `json:"format,omitempty"`
+	IncludeHeaders bool       `json:"include_headers,omitempty"`
+	Delimiter      rune       `json:"delimiter,omitempty"`
+}
+
+func (s *FileSpec) SetDefaults() {
+	if s.Delimiter == 0 {
```

Comment:
Delimiter can be the `NUL` character, so depending on the zero value doesn't do it 100% :)

Reply:
Seems like we're unlikely to run into this in practice, but if we want to be safe we could probably change Delimiter to `string` and use the first rune in the string (with validation that it doesn't have > 1 runes). Then we can check for the empty string here.